### ISSUE TITLE
POST response is not returning Navigator Objects for Rels

### ISSUE THREAD

**Issue by dudycooly** [deontologician/restnavigator#5]
POST response is returned as HAL, however Navigator is not created for POST Response HAL rels

```
In [238]: app_root['app:begin-flow'].expand(acct_id=14012670).create({})
Out[238]: (200, <Response [200]>)    <<<<<<<<<<<<<<<<<<  HALNavigator object expected here

In [239]: resp = app_root['app:begin-flow'].expand(acct_id=14012670).create({})

In [240]: resp[1].json()
Out[240]: 
{u'_links': {u'app2:portfolio': {u'href': u'http://server/account/14012670/portfolio',
   u'method': u'GET',
   u'templated': False},
  u'app1:get-info': {u'href': u'http://server/webapi/interaction/get_info/0aed14c105a44e5cbe83d65922db5e8a',
   u'method': u'GET'},
  u'app1:self': {u'href': u'http://self/webapi/interaction?account=14012670'},
  }}
```

Expected:
As in the README example:

```
>>> fred23 = N['ht:signup'].create(
... {'username': 'fred23',
...  'password': 'some_passwd',
...  'real_name': 'Fred 23'}
... )
>>> fred23
HALNavigator(haltalk.users.fred23)                          <<<<<<<<<<<<<<<<<< 
```

**Comment by deontologician** [comment]
Hmm, that's weird. I'll check it out

**Comment by deontologician** [comment]
OK, so the returning a tuple is really weird. I changed it so that it only returns the response directly. You can only get a HALNavigator from a POST if the response is a 201, 202, 301 or 302, since those have a `Location` header that corresponds to a real resource. The result of a post can't be represented as a resource since it's dependent on the initial POST data in the request (i.e. it isn't idempotent)

**Comment by deontologician** [comment]
@dudycooly I pushed a fix for this. Can you try building it and see if it works for you?

**Comment by dudycooly** [comment]
It is not returning tuple now,  however  it is still returning a  response object instead of HALNavigator

In [12]: app['app:begin-flow'].expand(acct_id=12670).create({})
Out[12]:  < Response [200] >

---

**Comment by deontologician** [comment]
@dudycooly It can't really return a Navigator since Navigators have to have a URL that defines them. What is the status code you're getting back from the POST? If it's redirecting you, you can get a Navigator for where it's redirecting you to

**Comment by dudycooly** [comment]
I nailed down this to

```
    if response.status_code in (httplib.CREATED,
                                httplib.ACCEPTED,
                                httplib.FOUND,
                                httplib.SEE_OTHER,
```

I have to change this into 

```
    if response.status_code in (httplib.CREATED,
                                httplib.ACCEPTED,
                                httplib.FOUND,
                                httplib.SEE_OTHER,
                                httplib.OK,  <<<<<<  to handle post response with 200
```

however i couldnt proceed due to the issue (#6)  with autofetch that I raised
If you think httplib.OK should be added, please do so and close this. thanks

**Comment by deontologician** [comment]
Do you have control over this api server? Basically, a 200 should return the result of the request, not redirect you to a new resource. It should be a 201 (Created) instead.

That being said, there should probably be some way for you to customize this behavior since you can't always talk to a server that follows http to the letter. I'll think about this and see what I come up with

**Comment by deontologician** [comment]
@dudycooly OK, I added a fix. Basically, a POST that results in a 200 will give you a new `PostResponse` object, which you can get the links from. The `PostResponse` is not fetchable, but you can go back to its parent (which is fetchable).

Example:

``` python
>>> N2 = N['x:create-object']
>>> post_response = N2.create({})
>>> post_response.links  # will work
>>> post_response() # will not work
>>> post_response.create({}) # will not work
>>> post_response.parent is N2
True
```

Let me know if this works for your use case

**Comment by deontologician** [comment]
I should note, that if the result of the POST is a 201, 202, 301 or 302, it will still give you a new `HALNavigator` created from the `Location` header (as it did before), instead of one of these `PostResponse` objects

**Comment by deontologician** [comment]
This should also solve #6

**Comment by dudycooly** [comment]
Tested and working fine. Thanks for the enhancement. It also partially  solved #6

**Comment by deontologician** [comment]
Awesome. Yeah I think this strategy is the best idea. There are just pseudo-resources in apis, and you still need to be able to interact with them even if they don't have a url. Thanks for poking me on this.
